SpecsUtils/src/pt/up/fe/specs/util/providers/impl/GenericFileResourceProvider.java

```diff
@@ -34,17 +34,14 @@ public static GenericFileResourceProvider newInstance(File file) {
      * 
      * <p>
      * Given file must exist, otherwhise an exception is thrown.
-     * 
-     * @param existingFile
-     * @param version
-     * @return
+     *
      */
     public static GenericFileResourceProvider newInstance(File existingFile, String version) {
         if (!existingFile.isFile()) {
             throw new RuntimeException("File '" + existingFile + "' does not exist");
         }
 
-        return new GenericFileResourceProvider(existingFile, version, false);
+        return new GenericFileResourceProvider(existingFile, version, version != null);
```

Comment:
I don't think this is equivalent, the isVersioned flag IIRC is related to the name of the resource (i.e., if it is expected that the version appears in the name). This is different from providing a version, which will be used to check if a new version should be used (regardless if the version is supposed to be in the name or not)